File: Javascript/easy/Exercism/perfect-numbers/perfect-numbers.js
export const classify = (number) => {
  if(number <= 0){ throw new Error('Classification is only possible for natural numbers.');}
  const classification = {
    perfect: 'perfect',
    abundant: 'abundant',
    deficient: 'deficient'
  };

  let aliquotSum = 0;
  for(let i = 1; i < number; i++){
    if(number % i === 0){ aliquotSum += i }
  }

  if(aliquotSum === number){
    return classification.perfect;
  } else if(aliquotSum > number){
    return classification.abundant;
  } else {
    return classification.deficient;
  }
};

File: Python/easy/Exercism/perfect-numbers/perfect_numbers.py
def classify(number):
    if number <= 0:
        raise ValueError("Classification is only possible for positive integers.")
    
    sum_divisors = sum(i for i in range(1, number) if number % i == 0)
    
    if sum_divisors == number:
        return "perfect"
    elif sum_divisors > number:
        return "abundant"
    else:
        return "deficient"


